bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/handler/VeluxHandler.java

```diff
@@ -125,4 +129,37 @@ public void handleConfigurationUpdate(Map<String, Object> configurationParameter
             super.handleConfigurationUpdate(configurationParameters);
         }
     }
+
+    /**
+     * Remove previously statically created vane channel if the device does not support it. Or log a warning if it does
+     * support a vane and the respective channel is missing.
+     *
+     * @param bridgeHandler the calling bridge handler.
+     * @throws IllegalStateException if something went wrong.
+     */
+    public void updateDynamicChannels(VeluxBridgeHandler bridgeHandler) throws IllegalStateException {
+        // roller shutters are the only things with a previously statically created vane channel
+        if (!VeluxBindingConstants.THING_TYPE_VELUX_ROLLERSHUTTER.equals(thing.getThingTypeUID())) {
+            return;
+        }
+
+        String id = VeluxBindingConstants.CHANNEL_VANE_POSITION;
+        ChannelUID uid = new ChannelUID(thing.getUID(), id);
+        Thing2VeluxActuator actuator = new Thing2VeluxActuator(bridgeHandler, uid);
+        VeluxProduct product = bridgeHandler.existingProducts().get(actuator.getProductBridgeIndex());
+
+        if (product.equals(VeluxProduct.UNKNOWN)) {
+            throw new IllegalStateException("initializeVanePositionChannel(): Product unknown in the bridge");
+        }
+
+        Channel channel = thing.getChannel(id);
+        boolean required = product.supportsVanePosition();
+
+        if (!required && channel != null) {
+            logger.debug("Removing unsupported channel for {}: {}", thing.getUID(), id);
+            scheduler.submit(() -> updateThing(editThing().withoutChannels(channel).build()));
```

Comment:
For my curiosity, why is it run in a separate thread ?

Reply:
Umm. When the bridge goes online, the bridge handler iterates over each of its things in a for loop with a call to this method, and I was a bit concerned about things updating themselves while the bridge is still within the process of updating _**itself**_, so I put it on a thread just in case. But maybe I am worrying about nothing?

Reply:
Seems similar to: https://github.com/openhab/openhab-addons/pull/13275#discussion_r947239063

Reply:
It is :)  ..
..  the only difference is that this time the method is called within the Bridge BridgeParamsUpdated method iteratively over all shades; so there are more bridges and things to possibly get entangled with each other..

Reply:
@jlaur: is it ok for you ?

Reply:
Ok. I have just tested it without the thread, and I confirm that there is no loop created.. So I will delete the submit() call.